test/org/sagebionetworks/bridge/stormpath/StormpathAccountDaoMockTest.java

```diff
@@ -318,28 +286,46 @@ public void updatingAccountWithRemovedGroups() {
 
     @Test(expected = IllegalArgumentException.class)
     public void getStudyPagedAccountsRejectsPageSizeTooSmall() {
-        StormpathAccountDao dao = new StormpathAccountDao();
         dao.getPagedAccountSummaries(study, 0, BridgeConstants.API_MINIMUM_PAGE_SIZE-1, null);
     }
 
     @Test(expected = IllegalArgumentException.class)
     public void getStudyPagedAccountsRejectsPageSizeTooLarge() {
-        StormpathAccountDao dao = new StormpathAccountDao();
         dao.getPagedAccountSummaries(study, 0, BridgeConstants.API_MAXIMUM_PAGE_SIZE+1, null);
     }
     
     @Test(expected = IllegalArgumentException.class)
     public void getStudyPagedAccountsRejectsNonsenseOffsetBy() {
-        StormpathAccountDao dao = new StormpathAccountDao();
         dao.getPagedAccountSummaries(study, -10, BridgeConstants.API_DEFAULT_PAGE_SIZE, null);
     }
-    
-    private SubpopulationService mockSubpopService() {
-        Subpopulation subpop = Subpopulation.create();
-        subpop.setGuidString(study.getIdentifier());
-        SubpopulationService subpopService = mock(SubpopulationService.class);
-        when(subpopService.getSubpopulations(study.getStudyIdentifier())).thenReturn(Lists.newArrayList(subpop));
-        return subpopService;
+
+    private com.stormpath.sdk.account.Account setupGroupChangeTest(Set<String> oldGroupSet, Set<Roles> newGroupSet) {
+        // mock Stormpath account
+        List<Group> mockGroupJavaList = new ArrayList<>();
+        for (String oneGroupName : oldGroupSet) {
+            Group mockGroup = mock(Group.class);
+            when(mockGroup.getName()).thenReturn(oneGroupName);
+            mockGroupJavaList.add(mockGroup);
+        }
+
+        GroupList mockGroupSpList = mock(GroupList.class);
+        when(mockGroupSpList.iterator()).thenReturn(mockGroupJavaList.iterator());
+
+        // Due to some funkiness in Stormpath's type tree, DefaultAccount is the only public class we can mock.
+        com.stormpath.sdk.account.Account mockSpAccount = mock(DefaultAccount.class);
+        when(mockSpAccount.getCustomData()).thenReturn(mock(CustomData.class));
+        when(mockSpAccount.getGroups()).thenReturn(mockGroupSpList);
+
+        // mock Bridge account
+        StormpathAccount mockAccount = mock(StormpathAccount.class);
+        when(mockAccount.getAccount()).thenReturn(mockSpAccount);
+        when(mockAccount.getRoles()).thenReturn(newGroupSet);
+
+        // execute
+        dao.updateAccount(mockAccount);
+
+        // return this, so the caller can verify back-end mock calls
+        return mockSpAccount;
     }
 
 }
```

Comment:
Test that the user does NOT have the health ID in the Stormpath account. The DAO should call the healthCodeService and set this up.